src/view/sidebar/calendar-components.jsx

```diff
@@ -0,0 +1,76 @@
+/*
+ * SPDX-FileCopyrightText: 2021 Zextras <https://www.zextras.com>
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+import React, { useCallback, useContext, useMemo } from 'react';
+import {
+	AccordionItem,
+	Button,
+	Container,
+	Dropdown,
+	Icon,
+	ModalManagerContext,
+	Padding,
+	Text,
+	Tooltip
+} from '@zextras/carbonio-design-system';
+import { useDispatch } from 'react-redux';
+import { FOLDERS } from '@zextras/carbonio-shell-ui';
+import { getShareInfo } from '../../store/actions/get-share-info';
+import { SharesModal } from './shares-modal';
+import { useCalendarActions } from '../../hooks/use-calendar-actions';
+
+export const SharesComponent = ({ item }) => {
+	const createModal = useContext(ModalManagerContext);
+	const dispatch = useDispatch();
+	const onClick = useCallback(
+		() =>
+			dispatch(getShareInfo()).then((res) => {
```

Comment:
we should create separate task to replace all these non required actions with proper promises because we are not creating any slice with getShareInfo so we can replace it with simple promise

Reply:
I found a very bad code base around in this place, i agree on changing them but I am not sure if we will have time to do it soon. Definitely we need to create issues to keep track of it for a proper time to clean the code